### DISCUSSION TITLE
Issues running Python on Hamilton

### DISCUSSION THREAD

**Discussion by barnabycollins** [wenceorg/comp4187#83] [Q&A] [answered]
Okay, so potentially stupid question incoming:

I've got myself all set up on Hamilton and have run:

```
module load gcc intelmpi openblas python
```

as written on the coursework spec. However, the command `python3.9` still doesn't work. I'm pretty sure this is because the newest Python install on Hamilton is 3.6.8; running `module avail` only lists 2.7.1, 2.7.9, 2.7.11, 2.7.13, 3.5.1 and 3.6.8. The command `module list` shows that `python/3.6.8` and `python/ompi/2.7.13` have been loaded.

So I've gone ahead with running `python3.6` instead, and it's at least running something, but now I'm getting the following error as soon as I try and do anything at all with it:

```
ModuleNotFoundError: No module named 'encodings'

Current thread 0x00002ac8e181d900 (most recent call first):
Aborted
```

Am I doing something wrong here? 😅 Also, just to check - does loading Python 3 override the `pip` command from the default Python 2.7 to the Python 3 environment, or should we use `python3.x -m pip ...` instead?

TIA!

**Comment by wence-** [comment, marked as answer]
Are you running on Hamilton 7 (`hamilton.dur.ac.uk`) which is end-of-life after Easter? Or Hamilton 8 (`hamilton8.dur.ac.uk`)? If the former, try switching to the latter.

As regards the pip question, you'll want to create a venv (with `python3.9 -m venv ...`), activate it, and then `pip` will be the one in your virtual environment.

**Reply by barnabycollins** [reply]
Hi - yeah, I was on Hamilton 7; I forgot they hadn't changed `hamilton.dur.ac.uk` to point to the new machines yet.

My bad; thank you!
